Library/Homebrew/unpack_strategy.rb

```diff
@@ -13,6 +13,7 @@ module UnpackStrategy
 
   def self.strategies
     @strategies ||= [
+      Crate, # Needs to be before Tar
```

Comment:
Need to think about this a bit more as `Crate` needs to be detected before `Tar`, but we don't want reverse.

May need to add explicit check on extension inside `Crate`.

Reply:
> * Could just allow `.crate` extension inside `Tar` instead. Simplest change as it only adds extra string to `unpack_strategy/tar.rb`.

If `.crate` is just a renamed `.tar.gz` then it makes send to do this.

Reply:
> If `.crate` is just a renamed `.tar.gz` then it makes sense to do this.

Currently it is though there is no documentation guaranteeing it. Assuming Rust doesn't switch to some obscure/specialized format, shouldn't be too difficult to maintain.

https://github.com/rust-lang/cargo/blob/0.78.0/src/cargo/ops/cargo_package.rs#L689-L694

Reply:
> Currently it is though there is no documentation guaranteeing it.

I think we can assume it is for now and then deal with it if/when things break.

Reply:
While it is an implementation detail, realistically a change to the format would likely break older `cargo` versions without branching off the API to download it.

Reply:
> I think we can assume it is for now and then deal with it if/when things break.

Agreed 👍🏻